### DISCUSSION TITLE
tcmalloc and jemalloc

### DISCUSSION THREAD

**Discussion by WolfBullet** [ImageMagick/ImageMagick#3220] [Help]
Hello

So I've been trying for some time now to find a way to add tcmalloc and if possible jemalloc (to compare) to ImageMagick. In the Threads of execution documentation, I found that you can add tcmalloc when I build the configure. So I'm now in visual studio in a command line and I am trying "msbuild configure.sln --with-tcmalloc". The only thing is the --with-tcmalloc is grayed out and when I try to build it, it doesn't recognize it. Do I have to install tcmalloc some way on to my windows or how can I add this to ImageMagick?

Thank you in advance

**Comment by dlemstra** [comment]
There is no build in support for using something like `tcmalloc` or `jemalloc` on Windows. You will probably need to add it yourself to the project that was created by `configure.exe`. We don't have any experience with this on Windows so we cannot help you out with this.

**Comment by WolfBullet** [comment]
OKe thank you for the info

Op zo 21 feb. 2021 om 10:39 schreef Dirk Lemstra <notifications@github.com>:

**Reply by slyfox1186** [reply]
I made this script to do what you are wanting. 

https://github.com/slyfox1186/script-repo

```
wget -qO imagick.sh https://imagick.optimizethis.net; sudo bash imagick.sh
```

Has to be done on a Linux Debian type OS. Try Ubuntu WSL. You can get it from the window store app.
